http/src/main/resources/swagger/api-docs.yaml

```diff
@@ -2909,6 +2909,9 @@ components:
           description: Optional environment variables to be set on the app
         auditInfo:
```

Comment:
updating jar should update swagger page

can you also list out the enum...appType isn't arbitrary string

Reply:
Hmm, it definitely didn't update the Swagger page.

appType isn't an arbitrary string, but if we list the values that it can be at the moment, we will have to keep updating it as new apps are added. Is that valuable enough to the user that it's worth the overhead and the possibility of being out of date? In my opinion, when you are looking at the what the types of apps in creation are, you don't really care about what other types they could have been.

Reply:
Maybe I could put something like "for example, GALAXY, CROMWELL, or CUSTOM" so that we aren't committed to updating it every time a new app is added?

Reply:
I agree with @Qi77Qi it'd be better to expose them as enums, and keep adding to the list as we go (since it doesn't happen often). This helps with request validation in Swagger.

Reply:
Btw I think we didn't add that validation here https://github.com/DataBiosphere/leonardo/blob/b647bfe08699f697a5a2cd3a1c43f2ffebda7419/http/src/main/resources/swagger/api-docs.yaml#L2866 - adding enum to requests is much more important than to responses.

Reply:
Just checked with @Qi77Qi , I think it's ok to keep as-is. The app creation request is getting validated on the backend, so there's no harm in keeping the response schema a little loose, from security perspective.

Reply:
huh...I think it's still helpful to use enum here regardless where the validation happens...If we were to add another apptype, that's sth should be made visiable, which updating swagger kinda servces the purpose as well

Reply:
That is exposed at the add App endpoint, because that is where you need to know the available options. I do not see why you need to know all the possible App options at the point where you are examining already created Apps.

Reply:
@cahrens I think @Qi77Qi is right on that from the API robustness principle perspective: https://www.programmableweb.com/news/why-following-robustness-principle-your-apis-might-be-good-idea/elsewhere-web/2017/05/20

Reply:
Basically, if our API response is lax and there're clients who start depending on the particular `appType` without us documenting them explicitly, that could mean if we later change the values of those `appType` (because they're not "in contract" if we don't declare them as such), then this could break existing clients.

That being said, not sure if folks are _actually_ using Leo APIs for any third-party integrations :-)

Reply:
I will look into creating a reference block can be referenced from addApp, listApp, and getApp. As long as I can avoid duplicating the list of app types I have no objections.

Reply:
Updated and tested on my fiab (I think I was experiencing caching issues previously).